lib/index.js

```diff
@@ -0,0 +1,169 @@
+require("dotenv").config();
+const { OpenAI } = require("openai");
+const fs = require("fs");
+const axios = require("axios");
+const cheerio = require("cheerio");
+const pretty = require("pretty");
+
+const OUTPUT_FOLDER = "summaries";
+
+const fileSystemSetup = () => {
+  if (!fs.existsSync(OUTPUT_FOLDER)) {
+    fs.mkdirSync(OUTPUT_FOLDER);
+  }
+
+  const dayString = new Date().toISOString().substring(0, 10);
+  if (!fs.existsSync(`${OUTPUT_FOLDER}/${dayString}`)) {
+    fs.mkdirSync(`${OUTPUT_FOLDER}/${dayString}`);
+  }
+
+  const hourString = new Date().toISOString().substring(11, 13);
+  if (!fs.existsSync(`${OUTPUT_FOLDER}/${dayString}/${hourString}`)) {
+    fs.mkdirSync(`${OUTPUT_FOLDER}/${dayString}/${hourString}`);
+  }
+
+  return { dayString, hourString };
+};
+
+function saveToFile(summary) {
+  summary = summary.trim();
+
+  const { dayString, hourString } = fileSystemSetup();
+
+  const rand = Math.floor(Math.random() * 1000000);
+  const timestamp = new Date().toISOString();
+  const fileName = `${OUTPUT_FOLDER}/${dayString}/${hourString}/${timestamp}-${rand}.md`;
+
+  fs.writeFile(fileName, summary, (err) => {
+    if (err) {
+      console.error("Error saving summary to file:", err);
+    } else {
+      console.info(`Summary saved to ${fileName} successfully.`);
+    }
+  });
+}
+
+async function fetchNews() {
```

Comment:
This function makes a call to a news API with a search query for "cats", sorted by the published date, and limited to only 10 results. We finally attract an API key at the end. Grab one from [News API](newsapi.org).